### PR TITLE
Build tools: update `latest` versions

### LINKED ISSUE #11671: Python 3.13 support
## What's the problem this feature will solve?

Python 3.13 is out in the wild but it is not currently possible with readthedocs to use it to genererate the documentation

image

## Describe the solution you'd like

Add python 3.13 to the available build tools

## Alternative solutions

We can of course use 3.12 in the meantime.

## Additional context

I noticed the problem when I tried bump python version to 3.13 in this project ( https://github.com/jrobichaud/django-structlog/pull/675) and the build failed.

### PR DESCRIPTION
**Pull request by laymonage**
Ref: #11386 and #10808

Not sure if I missed anything. Looking into `scripts/compile_version_upload.sh`, I don't know if the selected `virtualenv` version works with 3.13 (in fact I don't know if that package has officially supported 3.13 yet).

I initially wanted to raise an issue to support Node.js 22, but then I got curious and thought I could help speed up the process. Feel free to pick it up from here if there are other areas that need to be updated e.g. https://github.com/readthedocs/readthedocs-docker-images 🙂 

Closes  #11671

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11742.org.readthedocs.build/en/11742/

- Developer's documentation (`dev`): https://dev--11742.org.readthedocs.build/en/11742/

### CODE REVIEW

**Review by humitos — APPROVED** [review]
Thanks a lot for this PR. It looks great!

The only missing piece is another PR similar to https://github.com/readthedocs/readthedocs-docker-images/pull/205 in that repository. I'll see if I can make it now.

File: readthedocs/builds/constants_docker.py [hunk 1]
```diff
@@ -33,26 +33,30 @@
...
-            "16": "16.18.1",
-            "18": "18.16.1",
-            "19": "19.0.1",
-            "20": "20.14.0",  # LTS
+            "14": "14.21.3",
+            "16": "16.20.2",
+            "18": "18.20.4",
+            "19": "19.9.0",
+            "20": "20.18.0",  # LTS
+            "22": "22.11.0",  # LTS
```
Comment (humitos) [inline readthedocs/builds/constants_docker.py:55]:
```suggestion
            "20": "20.18.0",
            "22": "22.11.0",  # LTS
```

**Comment by humitos** [comment]
I opened https://github.com/readthedocs/readthedocs-docker-images/pull/209 with the last piece required to merge this PR.

**Comment by humitos** [comment]
We will need to update the examples from this file as well: https://github.com/readthedocs/ext-theme/blob/5a602534cb7f6252123d29e35c8e15b75c7a8350/readthedocsext/theme/templates/projects/import_config.html

**Comment by lb-** [comment]
Thanks all.

**Pull request merged**